internal/pkg/fleetapi/action.go

```diff
@@ -644,6 +648,61 @@ func (a *ActionApp) MarshalMap() (map[string]interface{}, error) {
 	return res, err
 }
 
+type ActionPrivilegeLevelChange struct {
+	ActionID   string                         `json:"id" yaml:"id"`
+	ActionType string                         `json:"type" yaml:"type"`
+	Data       ActionPrivilegeLevelChangeData `json:"data,omitempty"`
+
+	Err error `json:"-" yaml:"-" mapstructure:"-"`
+}
+
+// ID returns the ID of the Action.
+func (a *ActionPrivilegeLevelChange) ID() string {
+	return a.ActionID
+}
+
+// Type returns the type of the Action.
+func (a *ActionPrivilegeLevelChange) Type() string {
+	return a.ActionType
+}
+
+func (a *ActionPrivilegeLevelChange) String() string {
+	var s strings.Builder
+	s.WriteString("id: ")
+	s.WriteString(a.ActionID)
+	s.WriteString(", type: ")
+	s.WriteString(a.ActionType)
+	return s.String()
```

Comment:
Would it be useful to include `a.Data.String()` (after defining that method) as well?